packages/astro/src/core/render-context.ts

```diff
@@ -537,10 +537,10 @@ export class RenderContext {
 		// TODO: look into why computeCurrentLocale() needs routeData.route to pass ctx.currentLocale tests,
 		// and url.pathname to pass Astro.currentLocale tests.
 		// A single call with `routeData.pathname ?? routeData.route` as the pathname still fails.
-		return (this.#currentLocale ??=
-			computeCurrentLocale(routeData.route, locales, defaultLocale) ??
-			computeCurrentLocale(url.pathname, locales, defaultLocale) ??
-			fallbackTo);
+
+		return (this.#currentLocale ??= routeData.pathname
+			? computeCurrentLocale(routeData.pathname, locales, defaultLocale)
+			: (computeCurrentLocale(url.pathname, locales, defaultLocale) ?? fallbackTo));
```

Comment:
Should the `fallbackTo` apply to both the two results of the ternary? Right now it's only applying to if `route.pathname` is falsy.

Reply:
I think this is getting difficult to read. Can we convert this to boring `if/else` statements with assignment?